tests/test_coordinates.py

```diff
@@ -151,29 +156,24 @@ def test_setting_weights_invalid():
         sampling.weights = weights_invalid
 
 
-def test_quadrature_default_setter_getter():
-    """Test the default value, setter, and getter for quadrature."""
-
-    weights = [2 * np.pi, 2 * np.pi]
-    sampling = SamplingSphere([1, 1], 0, 0, weights=weights)
-
-    # test default value and getter
-    assert sampling.quadrature == False
-
-    # test setter and getter
-    sampling.quadrature = True
-    assert sampling.quadrature == True
-
-
-def test_quadrature_setter_errors():
-    """Test errors in the quadrature setter for wrong input data."""
-
-    sampling = SamplingSphere([1, 1], 0, 0)
-
-    # input type
-    with pytest.raises(TypeError, match="True or False but is None"):
-        sampling.quadrature = None
-
-    # weights are None
-    with pytest.raises(ValueError, match="quadrature can not be True"):
-        sampling.quadrature = True
+def test_quadrature_getter_changing_weights():
+    # create a quadrature grid (gaussian)
+    sampling = gaussian(n_max=3)
+    # check if quadrature is set properly
+    assert sampling.quadrature
+    # update weights such that quadrature requirement is not valid anymore
+    weights = 4 * np.pi * np.ones(sampling.cshape) / sampling.cshape
+    sampling.weights = weights
+    assert not sampling.quadrature
+
+
+def test_quadrature_getter_changing_points():
+    # create a quadrature grid (gaussian)
+    sampling = gaussian(n_max=3)
+    # check if quadrature is set properly
+    assert sampling.quadrature
+    # update points such that quadrature is not valid anymore
+    sampling.set_sph(np.random.rand(sampling.csize, 1),
+                     np.ones((sampling.csize, 1)),
+                     np.ones((sampling.csize, 1)))
```

Comment:
```suggestion
    sampling.spherical_colatitude = np.array(np.random.rand(sampling.csize, 1),
                     np.ones((sampling.csize, 1)),
                     np.ones((sampling.csize, 1)))
```
sth like this should work
I think this is deprecated and will be removed in 0.8.0.

Reply:
and fix ruff